cfme/common/vm.py

```diff
@@ -237,6 +239,25 @@ def compliant(self):
         else:
             raise ValueError("{} is not a known state for compliance".format(text))
 
+    @property
+    def console_handle(self):
+        '''
+        The basic algorithm for getting the consoles window handle is to get the
+        appliances window handle and then iterate through the window_handles till we find
+        one that is not the appliances window handle.   Once we find this check that it has
+        a canvas widget with a specific ID
+        '''
+        browser = self.appliance.browser.widgetastic
+        appliance_handle = browser.window_handle
+        cur_handles = browser.selenium.window_handles
+        logger.info("Current Window Handles:  {}".format(cur_handles))
+
+        for handle in cur_handles:
+            if handle != appliance_handle:
+                # XXX: Add code to verify the tab has the correct widget
```

Comment:
Currently, we have routine that closes the browser window/tab containing console after each test, if I ran it against 3 different providers back to back, irrespective of pass or fail on one of the providers, the next provider will not see any window other than appliance window. Hence this logic is currently not implemented. It is consistently working so far. 
I will put it FIXME for now.